plugin/pkg/scheduler/factory/plugins.go

```diff
@@ -99,6 +100,17 @@ func RegisterFitPredicate(name string, predicate algorithm.FitPredicate) string
 	return RegisterFitPredicateFactory(name, func(PluginFactoryArgs) algorithm.FitPredicate { return predicate })
 }
 
+// RegisterEssentialFitPredicate registers a fit predicate with the algorithm registry, the predicate is used by
+// kubelet, DaemonSet; it is always included in configuration. Returns the name with which the predicate was
+// registered.
+func RegisterEssentialFitPredicate(name string, predicate algorithm.FitPredicate) string {
```

Comment:
nit: I would probably use "mandatory" instead of "essential" to indicate that users cannot remove this predicate even if they want to.